cli/http_util.rs

```diff
@@ -14,6 +16,56 @@ use deno_runtime::deno_fetch::reqwest::Client;
 use deno_runtime::deno_fetch::reqwest::StatusCode;
 use std::collections::HashMap;
 
+#[derive(Debug, Clone)]
+struct AuthToken {
+  host: String,
+  token: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct AuthTokens(Vec<AuthToken>);
+
+impl AuthTokens {
+  pub fn new(maybe_token_str: Option<String>) -> Self {
+    let mut tokens = Vec::new();
+    if let Some(tokens_str) = maybe_token_str {
+      for token_str in tokens_str.split(';') {
+        if token_str.contains('@') {
+          let pair: Vec<&str> = token_str.split('@').collect();
```

Comment:
This means that the auth token could currently never include an `@` symbol. I think this is a reasonable constraint, but it could be fixed by only splitting on the last `@` as hostnames may never contain an `@` symbol.

Reply:
I will change it to `rsplitn`, safer that way